VotingApplication/VotingApplication.Web/Api/Controllers/ManageController.cs

```diff
@@ -189,16 +188,15 @@ public void Put(Guid manageId, ManagePollUpdateRequest updateRequest)
 
                 foreach (TokenRequestModel voter in updateRequest.Voters)
                 {
-                    if (voter.TokenGuid == null)
+                    if (!voter.EmailSent)
                     {
-                        Ballot newBallot = new Ballot { Email = voter.Email, TokenGuid = Guid.NewGuid() };
+                        Ballot newBallot = new Ballot { Email = voter.Email };
                         poll.Ballots.Add(newBallot);
-                        SendInvitation(poll.UUID, newBallot, poll.Name);
                     }
                     else
                     {
                         // Don't mark token as redundant if still in use
-                        Ballot ballot = redundantTokens.Find(t => t.TokenGuid == voter.TokenGuid);
+                        Ballot ballot = redundantTokens.Find(t => t.Email == voter.Email);
```

Comment:
As far as I am aware, emails aren't case sensitive. Maybe toLower() both of these?